### PR TITLE
[INFINITY-2977] Cassandra and Elastic Security Docs

### PR DESCRIPTION
**Pull request by benclarkwood**
This PR:
1. Depends on #2216.
2. Builds on the templating in #2216 and the prior PR for Kafka.
3. Adds a Cassandra security.md
4. Adds an Elastic security.md
5. Cleans up a lot of stuff in all 4 services
6. Fixes some dead links
7. And so much more!

### CODE REVIEW

File: docs/pages/_includes/services/security-transport-encryption-clients.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#### Transport Encryption Clients
+
+With Transport Encryption enabled, service clients will need to be configured to use [the DC/OS CA bundle](https://docs.mesosphere.com/latest/security/ent/tls-ssl/get-cert/) to verify the connections they make to the service. Consult your client's documentation for trusting a CA and configure your client appropriately.
```
Comment (elezar) [inline docs/pages/_includes/services/security-transport-encryption-clients.md]: Is it "**T**ransport **E**ncryption" or just "**t**ransport **e**ncryption"? I prefer the latter.

File: docs/pages/_includes/services/security-transport-encryption-clients.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#### Transport Encryption Clients
```
Comment (elezar) [inline docs/pages/_includes/services/security-transport-encryption-clients.md]: Transport Encryption **for** Clients?

File: frameworks/cassandra/docs/limitations.md [hunk 3]
```diff
@@ -19,7 +19,13 @@ The DC/OS {{ data.techName }} Service must be deployed with at least 3 nodes.
...
 ## Security
 
-{{ data.techName }}'s native TLS, authentication, and authorization features are not supported at this time.
+{{ data.techName }}'s native authentication, and authorization mechanisms are not supported at this time.
+
+### Toggling Transport Encryption
+
+Transport encryption using TLS can be toggled (enabled / disabled), but will trigger a rolling restart of the cluster. As each node restarts, a client may lose connectivity based on its security settings and the value of the `service.security.transport_encryption.allow_plaintext` configuration option. It is recommended that backups are made and downtime is scheduled.
+
+In order to enable TLS, a service account and corresponding secret is required. Since it is not possible to change the service account used by a service, it is recommended that the service is deployed with an explicit service account to allow for TLS to be enabled at a later stage.
```
Comment (elezar) [inline frameworks/cassandra/docs/limitations.md]: Should we link to service account docs?

File: frameworks/cassandra/docs/security.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+## Transport Encryption
+
+{% include services/security-transport-encryption-lead-in.md
+    techName="Apache Cassandra" plaintext="false" %}
+
+{% include services/security-configure-transport-encryption.md
+    techName="Apache Cassandra"
+    plaintext="true" %}
+
+*Note*: It is possible to update a running DC/OS Apache Cassandra service to enable transport encryption after initial installation, but the service may be unavilable during the transition. Additionally, your  clients will need to be reconfigured unless `service.security.transport_encryption.allow_plaintext` is set to `true`.
```
Comment (elezar) [inline frameworks/cassandra/docs/security.md]: There is some additional space between "your" and "clients".

Also, is it our intention to repeat the text from `limitations.md` here?

File: frameworks/elastic/docs/limitations.md [hunk 5]
```diff
@@ -20,8 +20,13 @@ The maximum number of deployable nodes is constrained by the DC/OS cluster's res
 
 Upgrades and rolling configuration updates do not wait for a green status. During deployment and upgrades, the `serial` strategy does not wait for the Elastic service to reach green before proceeding to the next node.
 
-## Toggling TLS
 
-Toggling TLS requires doing a full-cluster restart. This is an [Elasticsearch limitation](https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-tls.html). Since the service's default update strategy is a rolling upgrade (`serial`) changing its configuration to enable (or disable) transport encryption will result in nodes configured with TLS not being able to communicate with nodes configured with unencrypted networking (and vice-versa). A full-cluster restart is required (using the `parallel` update strategy). Make sure to have backups and plan for some downtime.
+## Security
+
+Elastic's native authentication and authorization mechanisms are not supported at this time.
```
Comment (mpereira) [inline frameworks/elastic/docs/limitations.md]: Should we clarify that "not supported" only means that it's not configurable from the service perspective? Authn and Authz are available and configurable when X-Pack is enabled.

Comment (benclarkwood) [reply frameworks/elastic/docs/limitations.md]: I'm going to leave it as the very broad "not supported" as we aren't supporting it directly, and the language leaves some wiggle room.

File: frameworks/cassandra/docs/security.md (hunk 4 above)
Comment (benclarkwood) [reply frameworks/cassandra/docs/security.md]: Yes, I want the security docs to fully document this limitation in place.

File: frameworks/cassandra/docs/limitations.md (hunk 3 above)
Comment (benclarkwood) [reply frameworks/cassandra/docs/limitations.md]: It is linked to in the relevant section in security.md. I think that's good enough.

File: docs/pages/_includes/services/security-transport-encryption-clients.md (hunk 1 above)
Comment (benclarkwood) [reply docs/pages/_includes/services/security-transport-encryption-clients.md]: I've been calling it Transport Encryption everywhere to make it more of a "thing"

**Pull request merged**